backend/site-api/src/db/custom_data_repo.rs

```diff
@@ -40,35 +39,85 @@ impl CustomDataRepoTrait for CustomDataRepo {
             .await
     }
 
-    async fn create_table(&self, dto: CreateTable) -> Result<(), Error> {
-        Ok(())
-    }
+    async fn create_table(&self, id: &str, dto: CreateTable) -> Result<(), DbError> {
+        let pool = self.get_db_pool(id).await?;
+
+        let table_name = dto.table_name;
+
+        let mut query: QueryBuilder<'_, Sqlite> = QueryBuilder::new("CREATE TABLE ");
+        query.push(table_name);
+        query.push(" (id INTEGER PRIMARY KEY NOT NULL");
+
+        for (column_name, column_info) in dto.columns.iter() {
+            let column_type = column_info.data_type.to_string();
+
+            query.push(format!(", {}", column_name));
+            query.push(" ");
+            query.push(column_type);
+            query.push(" NOT NULL");
+
+            for rule in column_info.validation_rules.iter() {
```

Comment:
This works, but we might eventually want to switch to validating in application code, instead of the database.

The reason is that since some validation must be implemented in the app, it's probably easier to maintain if we do everything that way. I think it also makes updating validation rules easier to reason about in some cases. That said, DB validation is probably more efficient, especially for `Unique`.

Reply:
Got it. So, should we keep these validations for now? Additionally, we'll also perform validations in the application code (if there are new validation rules added in the future, we'll add them to the application code).

Reply:
I think we can leave them in for now! I'll evaluate again before the feature is merged. After `ModifyColumn` is implemented, maybe it will be clearer whether the DB validations are too difficult to maintain.